src/vs/workbench/parts/extensions/electron-browser/extensionsActions.ts

```diff
@@ -1709,6 +1726,64 @@ export class ConfigureRecommendedExtensionsCommandsContributor extends Disposabl
 			},
 			when: this.workspaceFolderContextKey
 		});
+
+		CommandsRegistry.registerCommand(AddToWorkspaceRecommendationsAction.ID, serviceAccesor => {
+			this.boundAddToRecommendationsContextKey.set(false);
+			const extensionId = (this.editorService.activeEditor as ExtensionsInput).extension.id;
+			serviceAccesor
+				.get(IInstantiationService)
+				.createInstance(
+					AddToWorkspaceRecommendationsAction,
+					AddToWorkspaceRecommendationsAction.ID,
+					AddToWorkspaceRecommendationsAction.LABEL
+				)
+				.run(extensionId)
+				.then(
+					(quickPickCancelled) => this.boundAddToRecommendationsContextKey.set(quickPickCancelled),
+					() => this.boundAddToRecommendationsContextKey.set(this.editorService.activeEditor instanceof ExtensionsInput)
+				);
+		});
+		MenuRegistry.appendMenuItem(MenuId.CommandPalette, {
+			command: {
+				id: AddToWorkspaceRecommendationsAction.ID,
+				title: `${ExtensionsLabel}: ${AddToWorkspaceRecommendationsAction.LABEL}`
+			},
+			when: this.addToRecommendationsContextKey
+		});
+	}
+
+	private updateAddToRecommendationsCommandContextKey(): void {
+		if (!(this.editorService.activeEditor instanceof ExtensionsInput)) {
+			return this.boundAddToRecommendationsContextKey.set(false);
+		}
+
+		const folders = this.workspaceContextService.getWorkspace().folders;
+		const configurationFiles = folders
+			.map(workspaceFolder => workspaceFolder.toResource(paths.join('.vscode', 'extensions.json')));
+		const getFoldersRecommendedExtensionsPromises = configurationFiles.map(configurationFile => {
```

Comment:
one line arrow functions can lose the brackets and return function. ( `configurationFile => this.getFolderRecommendedExtensions(configurationFile)` does the same as this, but a bit more concisely).

Also, I'd look into just chaining these maps together to reduce the clutter of temporary variables.